### PR TITLE
Make it possible to build on Octave (Windows) - FIXED

### PR DESCRIPTION
**Pull request by rafaelgm**
I accidentally uploaded a version before where the 'octave' variable was not global, and that broke the script.

I'm very sorry, that was fixed now.

### CODE REVIEW

**Comment by amroamroamro** [comment]
Instead of using global variables (which I'm not really fond of), you should add another function with persistent variable to cache the result. Something like this: https://www.gnu.org/software/octave/doc/interpreter/How-to-distinguish-between-Octave-and-Matlab_003f.html

**Comment by rafaelgm** [comment]
I agree. Actually, before creating the variable, I was calling that exist(...) function all over the code. Then I changed to a variable and forgot there were many subfuncions (primary mistake).

I liked the function option you provided on the link. I'll edit the file using that, thanks!

File: +mexopencv/make.m [hunk 1]
```diff
@@ -436,3 +434,16 @@ function check_path_mexopencv(opts)
...
 end
+
+%%
+%% Return: true if the environment is Octave.
+%%
+function retval = isOctave
+    persistent cacheval;  % speeds up repeated calls
+
+    if isempty (cacheval)
+        cacheval = (exist ("OCTAVE_VERSION", "builtin") > 0);
```
Comment (amroamroamro) [inline +mexopencv/make.m]: you cant have double-quoted strings in MATLAB (only Octave), you gotta make them single-quotes...

Comment (rafaelgm) [inline +mexopencv/make.m]: Just noticed that and changed. Thanks.

2015-06-12 11:42 GMT-03:00 Amro <notifications@github.com>:

**Comment by kyamagu** [comment]
@amroamroamro and @rafaelgm Thanks!

**Comment by amroamroamro** [comment]
@rafaelgm: since you're here, can I ask if you've tested mexopencv in Octave 4 against OpenCV 2.x? Are all the class features working?

**Comment by rafaelgm** [comment]
@amroamroamro No, I haven't

**Pull request merged**